ion/src/shared/unicode/utf8_decoder.cpp

```diff
@@ -63,6 +63,33 @@ CodePoint UTF8Decoder::previousCodePoint() {
   return CodePoint(result);
 }
 
+const char * UTF8Decoder::nextGlyphPosition() {
+  assert(*m_stringPosition != 0 && (m_stringPosition == m_string || *(m_stringPosition - 1) != 0));
+  CodePoint followingCodePoint = nextCodePoint();
+  const char * resultGlyphPosition = m_stringPosition;
+  followingCodePoint = nextCodePoint();
+  while (followingCodePoint != UCodePointNull && followingCodePoint.isCombining()) {
+    resultGlyphPosition = m_stringPosition;
+    followingCodePoint = nextCodePoint();
+  }
+  m_stringPosition = resultGlyphPosition;
+  return resultGlyphPosition;
+}
+
+const char * UTF8Decoder::previousGlyphPosition() {
+  assert(m_stringPosition > m_string);
+  CodePoint previousCP = previousCodePoint();
+  if (m_stringPosition == m_string) {
```

Comment:
Why do you need this special case? Won't it work the same without?

Reply:
That's right :)